bundles/io/org.openhab.io.myopenhab/src/main/java/org/openhab/io/myopenhab/internal/MyOpenHABServiceImpl.java

```diff
@@ -0,0 +1,289 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.io.myopenhab.internal;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.openhab.io.myopenhab.MyOpenHABService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.library.items.RollershutterItem;
+import org.openhab.core.library.items.SwitchItem;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.scriptengine.action.ActionService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.TypeParser;
+import org.openhab.core.persistence.PersistenceService;
+import org.openhab.ui.items.ItemUIRegistry;
+
+/**
+ * This class starts my.openHAB connection service and implements interface to communicate with my.openHAB.
+ * It also acts as a persistence service to send commands and updates for selected items to my.openHAB
+ * and processes commands for items received from my.openHAB.
+ * 
+ * @author Victor Belov
+ * @since 1.3.0
+ *
+ */
+
+public class MyOpenHABServiceImpl implements MyOpenHABService, PersistenceService, ManagedService, ActionService, MyOHClientListener {
```

Comment:
i would prefer splitting this class into it's different functionalities (PersistenceService, ActionService and ServiceImpl)